src/fs/filesystem.ts

```diff
@@ -191,6 +197,7 @@ export class FileSystem {
     const head = parts[0]
     const relPath = pathUtil.join(parts.slice(1))
 
+    // Why `a` not FileSystem? TS has subtyping. I easily could be missing something.
```

Comment:
Methods return Trees, Files, CIDs, FileContent, etc.  This generic is just "whatever the method returns"

Note: I'm not very happy with this function, and it's getting more complex as the code grows. I'm definitely looking for ways to improve it

Reply:
Okay cool cool! Thanks 😄